ch13/ex13_30.h

```diff
@@ -18,7 +18,14 @@
 
 class HasPtr {
 public:
-    friend void swap(HasPtr&, HasPtr&);
+    friend inline void swap(HasPtr& lhs, HasPtr& rhs)
+    {
+    using std::swap;
+    swap(lhs.ps, rhs.ps);
+    swap(lhs.i, rhs.i);
+    std::cout << "call swap(HasPtr& lhs, HasPtr& rhs)" << std::endl;
+    }
+    
     HasPtr(const std::string &s = std::string()) : ps(new std::string(s)), i(0) { }
```

Comment:
@Soyn  Please keep the **indentation**, and the `inline` there is **redundant**.